### PR TITLE
Convert datasets to av1 encoding

### PR DESCRIPTION
**Pull request by aliberts**
## What this does
Adds a small script to easily convert all of our datasets to AV1 encoding once #282 is done.
This will be run with:
```bash
python lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py \
  --raw-dir /fsx/simon_alibert/lerobot/data/raw \
  --raw-repo-ids lerobot-raw \
  --local-dir /fsx/simon_alibert/lerobot/data/push \
  --tests-data-dir tests/data \
  --push-repo lerobot \
  --vcodec libsvtav1 \
  --pix-fmt yuv420p \
  --g 2 \
  --crf 30
```

It also adds the encoding information to the meta-data `info.json`:
```diff
{
    "codebase_version": "v1.6",
    "fps": 50,
    "video": true,
+   "encoding": {
+       "vcodec": "libsvtav1",
+       "pix_fmt": "yuv420p",
+       "g": 2,
+       "crf": 30
+   }
}
```

This bumps up `CODEBASE_VERSION`: `v1.5` -> `v1.6`.

## How it was tested
Policies have been trained and evaluated on AV1-encoded datasets and compared against the current reference (h264):
- Diffusion on pusht:
  - [h264-encoded run](https://wandb.ai/aliberts/lerobot/runs/zubx2fwe/workspace?nw=nwuseraliberts)
  - [AV1-encoded run](https://wandb.ai/aliberts/lerobot/runs/mlestyd6/workspace?nw=nwuseraliberts)
- ACT on aloha_sim_transfer_cube_human:
  - [h264-encoded run](https://wandb.ai/aliberts/lerobot/runs/le8scox9?nw=nwuseraliberts)
  - [AV1-encoded run](https://wandb.ai/aliberts/lerobot/runs/rz454evx/workspace?nw=nwuseraliberts)
- ACT on aloha_sim_insertion_scripted:
  - [h264-encoded run](https://wandb.ai/aliberts/lerobot/runs/r6q2bsq4/workspace?nw=nwuseraliberts)
  - [AV1-encoded run](https://wandb.ai/aliberts/lerobot/runs/4abyvtcz/workspace?nw=nwuseraliberts)

For more info on AV1 encoding and its performance, have a look at the results from #282.

Example dataset: [aliberts/aloha_sim_insertion_human](https://huggingface.co/datasets/aliberts/aloha_sim_insertion_human)
This has been added with:
```bash
python lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py \
  --raw-dir data \
  --raw-repo-ids \
    cadene/aloha_sim_insertion_human_raw \
  --push-repo aliberts \
  --vcodec libsvtav1 \
  --pix-fmt yuv420p \
  --g 2 \
  --crf 30
```

## How to checkout & try? (for the reviewer)
```bash
mkdir -p data/dummy_repo/dummy && \
python lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py \
  --raw-dir data \
  --raw-repo-ids dummy_repo/dummy \
  --push-repo dummy_repo \
  --raw-format aloha_hdf5 \
  --dry-run 1
```

### CODE REVIEW

File: lerobot/common/datasets/lerobot_dataset.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 from lerobot.common.datasets.video_utils import VideoFrame, load_from_videos
 
 DATA_DIR = Path(os.environ["DATA_DIR"]) if "DATA_DIR" in os.environ else None
-CODEBASE_VERSION = "v1.5"
+CODEBASE_VERSION = "v1.6"
```
Comment (Cadene) [inline lerobot/common/datasets/lerobot_dataset.py:39]: I was suggestion v2.0 but it's too late I think

File: lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    raw_repo_ids: list[str],
+    push_repo: str,
+    vcodec: str,
+    pix_fmt: str,
+    g: int,
+    crf: int,
+    dry_run: bool = False,
+) -> None:
+    if len(raw_repo_ids) == 1 and raw_repo_ids[0].lower() == "all":
+        raw_repo_ids = [f"cadene/{id_.split('/')[1]}_raw" for id_ in available_datasets]
```
Comment (Cadene) [inline lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py:27]: Make it configurable maybe?
```suggestion
        raw_repo_ids = [f"{hf_user}/{id_.split('/')[1]}_raw" for id_ in available_datasets]
```

File: lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    for dataset_repo_id_raw in raw_repo_ids:
+        dataset_id_raw = dataset_repo_id_raw.split("/")[1]
+        dataset_id = dataset_id_raw[:-4]
+        dataset_repo_id_push = f"{push_repo}/{dataset_id}"
+        raw_format = SOURCE_FORMAT[dataset_id.split("_")[0]]
```
Comment (Cadene) [inline lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py:33]: Tricky. Would be nice to have a more future proof way than `SOURCE_FORMAT[dataset_id.split("_")[0]]`

Maybe by listing all the raw dataset somewhere associated with their format?

Or maybe to have a dedicated function `get_raw_format(raw_dataset_id)`?

Maybe fine for now

**Comment by Cadene** [comment]
Hey @enzostvs @haixuanTao , be aware that we have updated our encoding format. I remember Dora-LeRobot uses our encoding function, but not sure.
Thanks!
cc @aliberts

**Review by Cadene — APPROVED** [review]
LGTM, left a few comments

File: lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py (hunk 3 above)
Comment (aliberts) [reply lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py:33]: Working on it to come up with a better way.

Comment (aliberts) [reply lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py]: Fixed https://github.com/huggingface/lerobot/pull/302/commits/fc3792d6ed290fa2f950221b76e104f6cd9dd5e0

File: lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py (hunk 2 above)
Comment (aliberts) [reply lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py]: UPDATE:
```diff
parser.add_argument(
        "--raw-repo-ids",
        type=str,
        nargs="*",
        default=["lerobot-raw"],
-       help="Dataset repo ids. if 'all', the list from `available_datasets` will be used.",
+       help="""Raw dataset repo ids. if 'lerobot-raw', the keys from `AVAILABLE_RAW_REPO_IDS` will be
+           used and raw datasets will be fetched from the 'lerobot-raw/' repo and pushed with their
+           associated format. It is assumed that each dataset is located at `raw_dir / raw_repo_id` """,
    )
```

Comment (aliberts) [reply lerobot/common/datasets/push_dataset_to_hub/_convert_datasets.py]: (I don't handle downloading raw datasets in this script anymore as it should be done outside)

File: lerobot/common/datasets/lerobot_dataset.py (hunk 1 above)
Comment (Cadene) [reply lerobot/common/datasets/lerobot_dataset.py:39]: Sounds good

File: lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
+import argparse
+from pathlib import Path
+
```
Comment (Cadene) [inline lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py:3]: Could you add a docstring at the top of the file to explain what is it useful for and add an example usage (with the command your ran for instance)

**Review by Cadene — APPROVED** [review]
Very cool. Added a minor comment.

File: lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py (hunk 4 above)
Comment (aliberts) [reply lerobot/common/datasets/push_dataset_to_hub/_encode_datasets.py:39]: Done https://github.com/huggingface/lerobot/pull/302/commits/c82c11678b66e11ce763e2474479b324696ca93f

**Pull request merged**